### DISCUSSION TITLE
Multiprocessing on TPU from colab

### DISCUSSION THREAD

**Discussion by kallel-mahdi** [jax-ml/jax#13634] [Q&A]
I am running compute-heavy machine learning experiments with multiple seeds.

The code contains non-functional segments preventing me from using PMAP to parallelize across seeds.

I am currently using google Colab to access TPU's

Is there a way to launch a different process running on a separate TPU core for every seed?

I tried to follow the advice in this [blog](https://blog.jamiebernardi.com/2021/12/12/how-to-run-concurrent-processes-on-multiple-devices-in-jax/) 

However, it does not work on colab as I get the same device for all processes.

I created a [notebook](https://colab.research.google.com/gist/kallel-mahdi/2970213334d897d00129ac19277b1b5b/jax_tpu_multiprocessing.ipynb) to reproduce the issue.

**Comment by functionstackx** [comment]
Hi @kallel-mahdi !

I believe that the running 1 process per each tpu core requires a tpu vm and not a tpu core.

any reason why you would want to use multiple proccesses? instead of creating an array of keys then passing that into a pmap function to parallelize across all the cores?

**Reply by kallel-mahdi** [reply]
Hi @OrenLeung ,

Thank you for your response.

On google Colab we have access to TPU v2-8 instances that have 4 cores. 

I use weights&biases for logging, having a separate process for every seed will simplify my experiment tracking.

Many thanks

**Reply by functionstackx** [reply]
yep! i believe the feature is not supported on colab :(
